src/main/java/org/jvnet/hudson/test/TemporaryDirectoryAllocator.java

```diff
@@ -115,13 +110,34 @@ public synchronized void disposeAsync() {
 
         new Thread("Disposing "+base) {
             public void run() {
-                for (File dir : tbr)
+                for (File dir : tbr) {
+                    LOGGER.info(() -> "deleting " + dir);
                     try {
-                        new FilePath(dir).deleteRecursive();
-                    } catch (IOException | InterruptedException e) {
-                        e.printStackTrace();
+                        delete(dir.toPath());
+                    } catch (IOException e) {
+                        LOGGER.log(Level.WARNING, null, e);
                     }
+                }
             }
         }.start();
     }
+
+    private void delete(Path p) throws IOException {
+        LOGGER.fine(() -> "deleting " + p);
+        if (Files.isDirectory(p, LinkOption.NOFOLLOW_LINKS)) {
```

Comment:
probably also a junction which I think may be used in some test case and `FIlePath` respects.. (`FIles.readAttributes` and `isOther` - as a general well it is not a normal directory)

Reply:
I do not know much about junctions. Would they ever really be used in `$JENKINS_HOME`?

Reply:
I'm sure something uses them as we have specific code in Jenkins...
https://github.com/jenkinsci/jenkins/blob/715cdb15b3e855f8b6115aa02200860ed6caf49d/core/src/main/java/hudson/Util.java#L323-L343
I have not come across a real life example, but I'm guessing that it is for a reason?

Maybe it is just an easy way in Jenkins to put some subset of `Job`s in a different drive / pool etc and would not actually be done *inside* a workspace, or a build directory by jenkins (I do not know if zipping on windows copies the Junction or not, so if a workspace was coppied from an agent where the build created a junction if that would be replcicated on a master).

Reply:
There might be some obscure use case in production servers, but I doubt this would affect functional tests. If there is some plugin which does create a junction during a test for some reason, then it can explicitly remove it before generic cleanup, right?